iceoryx_posh/include/iceoryx_posh/runtime/service_discovery.hpp

```diff
@@ -18,12 +18,18 @@
 #define IOX_POSH_RUNTIME_SERVICE_DISCOVERY_HPP
 
 #include "iceoryx_posh/iceoryx_posh_types.hpp"
+#include "iceoryx_posh/internal/roudi/service_registry.hpp"
+#include "iceoryx_posh/popo/subscriber.hpp"
 #include "iceoryx_posh/runtime/posh_runtime.hpp"
 
 namespace iox
 {
 namespace runtime
 {
+enum class ServiceDiscoveryEvent : popo::EventEnumIdentifier
+{
+    SERVICE_REGISTRY_HAS_CHANGED
```

Comment:
Please rename it into `SERVICE_REGISTRY_UPDATED` ... `HAS_CHANGED` describes a state and the other one sounds like an event. See the subscriber state `HAS_DATA` and the event `DATA_RECEIVED`.